File: Part1_강의자료/Ch03_배열/문제별코드/1236_성지키기/1236.cpp
#include <algorithm>
#include <stdio.h>

using namespace std;

char castle[55][55];
int N, M;

int is_exist(int x, int y, int dx, int dy, int end)
{
	for (int i = 0; i < end; ++i)
	{
		if (castle[y][x] == 'X')
			return 0;
		x += dx;
		y += dy;
	}
	return 1;
}

int main()
{
	scanf(" %d %d", &N, &M);
	for (int i = 0; i < N; ++i)
		scanf(" %s", castle[i]);

	int row = 0;
	for (int i = 0; i < N; ++i)
		row += is_exist(0, i, 1, 0, M);
	
	int col = 0;
	for (int i = 0; i < M; ++i)
		col += is_exist(i, 0, 0, 1, N);
	
	printf("%d\n", max(row, col));
	return 0;
}

File: Part1_강의자료/Ch03_배열/문제별코드/1236_성지키기/1236.py
N, M = map(int, input().split())
castle = [[0] * M for _ in range(N)]
row = 0
col = 0

for i in range(N):
	line = input()
	cnt = 0
	for j in range(M):
		castle[i][j] = line[j]
		if line[j] == 'X':
			cnt += 1
	if cnt == 0:
		col += 1

for i in range(M):
	if 'X' not in [castle[j][i] for j in range(N)]:
		row += 1

print(max(row, col))